collector/exporter.go

```diff
@@ -147,17 +156,17 @@ func (e *Exporter) scrape(ctx context.Context, ch chan<- prometheus.Metric) {
 
 	if err := db.PingContext(ctx); err != nil {
 		level.Error(e.logger).Log("msg", "Error pinging mysqld", "err", err)
-		e.metrics.MySQLUp.Set(0)
-		e.metrics.Error.Set(1)
+		ch <- prometheus.MustNewConstMetric(mysqlUp, prometheus.GaugeValue, 0.0)
+		ch <- prometheus.MustNewConstMetric(mysqlScrapeFailed, prometheus.GaugeValue, 0.0)
```

Comment:
I wonder if we should just get rid of `mysql_last_scrape_failed`. It's basically the same thing as `mysql_up`.